### PR TITLE
Camera-Ready Artefact Submission

### PR DESCRIPTION
**Pull request by pmbittner**
As watching the diff here might not be too helpful, you can find the current readme in beautiful here: https://github.com/pmbittner/FeatureTraceRecording/tree/cameraready

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,21 +19,53 @@ The artefact mainly consists of a library written in the Haskell language that i
...
+Assume there is a second developer, Bob, working on another variant (e.g., branch or fork) of the software.
+Bob’s variant might implement other features than Alice’s.
+Assume Bob’s variant implements the blue feature <span style="color:#00BEEB">SafeStack</span> but not the orange feature <span style="color:#D95F02">ImmutableStack</span>.
+Although Alice made many edits to <span style="color:#D95F02">ImmutableStack</span>, we can still derive feature mappings for Bob’s code.
+When Alice deletes code from <span style="color:#D95F02">ImmutableStack</span>, we can infer that this code does not belong to <span style="color:#D95F02">ImmutableStack</span> anymore.
+We thus introduce the new feature mapping <span style="color:#7570B3">¬ImmutableStack</span> for the deleted code that is still present in Bob’s variant (highlighted in purple):
+
+<img src="meta/BobsVariant.png" alt="Bob's Recorded Feature Traces" width="250" />
+
+A detailed explanation of this example can be found in our [paper][paper] and [preprint][preprint].
```
Comment (AlexanderSchultheiss) [inline README.md:46]: The link for `paper` is leading to a DOI not found page, but the DOI of the paper has just not been enabled yet, right?

File: README.md [hunk 2]
```diff
@@ -15,21 +19,53 @@ The artefact mainly consists of a library written in the Haskell language that i
...
+    - [`userFormat`](https://pmbittner.github.io/FeatureTraceRecording/docs/Main.html#v:userFormat) (default): The perspective of the developer who is editing source code while traces are recorded in the background. This is the format used in the figures in the paper. The tool will show the presence conditions of the snapshots.
+    - [`userFormatDetailed`](https://pmbittner.github.io/FeatureTraceRecording/docs/Main.html#v:userFormatDetailed): A variation of `userFormat` where traces and presence conditions can be investigated seperately at the same time. Code is coloured in the colour of its feature trace while presence conditions are indicated by coloured lines on the left.
+    - [`astFormat`](https://pmbittner.github.io/FeatureTraceRecording/docs/Main.html#v:astFormat): Shows the abstract syntax tree of the source code with feature traces as formulas.
+    - [`tikzFormat`](https://pmbittner.github.io/FeatureTraceRecording/docs/Main.html#v:tikzFormat): Tikz export of abstract syntax trees with traces. Used for figures in the paper.
```
Comment (AlexanderSchultheiss) [inline README.md:66]: Maybe add where they are exported to

File: README.md [hunk 3]
```diff
@@ -44,7 +80,37 @@ Some interesting code locations are:
...
+	title = {{Feature Trace Recording}},
+	booktitle = {Proc.\ Europ.\ Software Engineering Conf./Foundations of Software Engineering (ESEC/FSE)},
+	location = {Athens, Greece},
+	publisher = {ACM},
+	address = {New York, NY, USA},
+	year = 2021,
+	month = AUG,
+	note = {To appear}
+}
+```
```
Comment (AlexanderSchultheiss) [inline README.md:99]: Shouldn't the citation refer to the artifact instead of the paper? Maybe add the URL or Zenodo DOI?

File: README.md [hunk 4]
```diff
@@ -44,7 +80,37 @@ Some interesting code locations are:
...
+Instead, users of this library have to provide [Abstract Syntax Trees (ASTs)](https://pmbittner.github.io/FeatureTraceRecording/docs/AST.html) and [edits to them](https://pmbittner.github.io/FeatureTraceRecording/docs/Edits.html) as input for feature trace recording (see `runFTR` and `runFTRWithIntermediateSteps` in [FeatureTraceRecording.hs](https://pmbittner.github.io/FeatureTraceRecording/FeatureTraceRecording.html)).
+Examples for creating edits and ASTs can be found in the demos for Alice ([docs](https://pmbittner.github.io/FeatureTraceRecording/docs/StackPopAlice.html), [src](app/examples/StackPopAlice.hs)), Bob ([docs](https://pmbittner.github.io/FeatureTraceRecording/docs/StackPopBob.html), [src](app/examples/StackPopBob.hs)), and edit patterns ([docs](https://pmbittner.github.io/FeatureTraceRecording/docs/EditPatterns.html), [src](app/examples/EditPatterns.hs)).
+
+Currently, the library also does not provide (de-)serialisation of feature traces.
+
+
+## Contact
+
+Don't hesitate to open issues or pull-request or to contact us directly (paul.bittner@uni-ulm.de). We are thankful for any questions, constructive criticism, or interest. :blush:
```
Comment (AlexanderSchultheiss) [inline README.md:113]:
```suggestion
Don't hesitate to open issues or pull-request, or to contact us directly (paul.bittner@uni-ulm.de). We are thankful for any questions, constructive criticism, or interest. :blush:
```

File: README.md [hunk 5]
```diff
@@ -54,4 +120,8 @@ You can inspect the truth tables for the ternary logic by uncommenting the respe
 [jeffrey]: https://www.uni-ulm.de/in/sp/team/former-employees-and-doctorands/jeffrey-young/
 [lukas]: https://www.tu-braunschweig.de/isf/team/lukas-linsbauer
 
-[documentation]: https://pmbittner.github.io/FeatureTraceRecording/
\ No newline at end of file
+[website]: https://pmbittner.github.io/FeatureTraceRecording/
+[documentation]: https://pmbittner.github.io/FeatureTraceRecording/docs
+[preprint]: https://github.com/SoftVarE-Group/Papers/raw/master/2021/2021-ESECFSE-Bittner.pdf
+[paper]: https://doi.org/10.1145/3468264.3468531
+[variantsync]: https://www.uni-ulm.de/en/in/sp/research/projects/variantsync/
```
Comment (AlexanderSchultheiss) [inline README.md:127]: Cool! I didn't know that it is possible to specify variables

File: README.md [hunk 6]
```diff
@@ -15,21 +19,53 @@ The artefact mainly consists of a library written in the Haskell language that i
 The library is accompanied with a demo application that uses the library to reproduce our motivating example (Alice and Bob using feature trace recording in Section 2 in our paper) as well as examples of the edit patterns we used to evaluate feature trace recording (Section 5).
 
 
+## What is Feature Trace Recording?
+
+Feature trace recording is a semi-automated method for recording feature-to-code mappings during software development.
+It unlocks evolution as a new source of information for feature location.
+
+![Feature Trace Recording Example](meta/FeatureTraceRecording.png)
```
Comment (AlexanderSchultheiss) [inline README.md:27]: The image is rendered really badly for me. It is kind of blurry and hurts the eyes. Strangely, everything appears to be fine when opening the image. 
Screenshot_20210716_145059

**Review by AlexanderSchultheiss — APPROVED** [review]
Only minor comments

File: README.md (hunk 1 above)
Comment (pmbittner) [reply README.md:46]: yes. This is the DOI ACM gave us. The paper should be available under this DOI once its published.

File: README.md (hunk 2 above)
Comment (pmbittner) [reply README.md:66]: to terminal :D I will add a note.

File: README.md (hunk 3 above)
Comment (pmbittner) [reply README.md:99]: We don't want people to cite the artefact but not the paper. Citing artefact and paper separately is weird, too. In any case, the paper is the primary artifact that should be cited. Also I never saw dedicated citations for only the artefacts of a paper but not the paper. I will add the DOI to the article.

File: README.md (hunk 6 above)
Comment (pmbittner) [reply README.md:27]: So this seems to be an issue with the markdown viewer you use. Which one was it? It looks fine for me in VSCode and on GitHub.

Comment (pmbittner) [reply README.md:27]: Also I have no idea how to change that as the image is fine as you said.

Comment (AlexanderSchultheiss) [reply README.md:27]: It was on GitHub using Google Chrome

Comment (pmbittner) [reply README.md:27]: Could be fixed by creating custom HTML to integrate the image... I have no idea how though and it that would work. Sounds like a general bug in Github+Chrome. :(

Comment (pmbittner) [reply README.md]: I tried a fix, please recheck.

Comment (pmbittner) [reply README.md]: ok fixed

**Pull request merged**